subprojects/execution/src/main/java/org/gradle/internal/execution/ReasonedContext.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution;
+
+import java.util.Optional;
+
+public interface ReasonedContext extends Context {
```

Comment:
Can we call this something else? `ContextWithRebuildReason` comes to mind.

Reply:
For now I'd like to keep the `Context` suffix. Will rename it to `ExecutionRequestContext`.